[docs] Add a Local app production and create a Local app section

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Based on feedback from @brentvatne on a missing in-detailed guide that covers steps on how to create a code signed production build for Android and iOS.

This new guide can be backlinked from other documents (like Deploy > Build a project from https://github.com/expo/expo/pull/31674) help us avoid explaining these steps. It also covers some Gradle specific configuration for Android which slightly differs from React Native's Android guide on the same subject.

# How

<!--
How did you build this feature or fix this bug and why?
-->

This PR:
- Creates a sub-section under Guides > Development Process called "Local app"
- Adds a new guide named "Create a local production build locally"
  - This guide covers steps to create a code signed production build for Android and iOS based on the steps from React Native's [Android](https://reactnative.dev/docs/signed-apk-android) and [iOS](https://reactnative.dev/docs/publishing-to-app-store) guides but cover the differences for an Expo project (only applied to Android). iOS steps are the same.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

See preview.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs] Add a Local app production and create a Local app section

amandeepmittal committed:
Update local-app-production.mdx

amandeepmittal committed:
Update local-app-production.mdx

brentvatne commented on docs/pages/guides/local-app-production.mdx:
```diff
@@ -0,0 +1,159 @@
+
+<Step label="1">
+
+### Create an upload key
```
we might want to add a note here - if you have previously used EAS Build to build your app, you should run `eas credentials -p android` in your project and download the app signing credentials - choose the build profile and then "Download credentials from EAS to credentials.json". The **credentials.json** file keystore password, key alias, and key password, and you can use those in the following steps instead of generating as keystore.

amandeepmittal replied:
Thanks for the suggestion and the explanation. Decided to add a collapsible for this at the beginning of the step like this:

![CleanShot 2024-10-07 at 00 57 13@2x](https://github.com/user-attachments/assets/9c6c34dc-4727-466b-bcb8-b987942a95cb)

amandeepmittal committed:
[docs] Add a section about local apps (#31745)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

Follow up #31743

Based on a user feedback on how to compile an app locally for
development and production.

The existing "Note" in this guide already covered the local development
process, but we didn't have steps to link for production. Based on the
changes from the PR linked above, we can add a section here to provide
context and links to both of the local app guides.

# How

<!--
How did you build this feature or fix this bug and why?
-->

- Remove the note at the top of the page
- Add a section: App compilation for development and production builds
locally

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

![CleanShot 2024-09-30 at 16 39
44@2x](https://github.com/user-attachments/assets/75f6d29c-a48e-409e-b5fc-9db0c283e31d)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

amandeepmittal committed:
Apply feedback from Brent's review
